### PR TITLE
Added some error handling in gcp cloud function scheduling

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
To be tested, but added the basic code for now

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/gcp/google_cloud_function.py [hunk 1]
```diff
@@ -174,5 +174,11 @@ def create_cloud_function(
         if time.time() - start_time > timeout:
             raise TimeoutError("Timed out waiting for function to deploy!")
 
+        if state == Function.State.FAILED:
+            error_messages = ', '.join([msg.text for msg in response.state_messages])
```
Comment (htahir1) [inline src/zenml/integrations/gcp/google_cloud_function.py:178]:
```suggestion
            error_messages = ', '.join([msg.message for msg in response.state_messages])
```

File: src/zenml/integrations/gcp/google_cloud_function.py [hunk 2]
```diff
@@ -174,5 +174,11 @@ def create_cloud_function(
         if time.time() - start_time > timeout:
             raise TimeoutError("Timed out waiting for function to deploy!")
 
+        if state == Function.State.FAILED:
+            error_messages = ', '.join([msg.message for msg in response.state_messages])
```
Comment (htahir1) [inline src/zenml/integrations/gcp/google_cloud_function.py:178]:
```suggestion
            error_messages = ", ".join([msg.message for msg in response.state_messages])
```

**Review by bcdurak — CHANGES_REQUESTED** [review]
I made a suggestion by adding a small adjustment to the if condition.

**Review by bcdurak — APPROVED** [review]

**Review by stefannica — APPROVED** [review]

**Comment by htahir1** [comment]
@bcdurak Could you kindly test this locally and merge if it works out as we expect?

**Comment by bcdurak** [comment]
@htahir1 on it!

**Pull request merged**